management/guardian_test.go

```diff
@@ -265,36 +265,80 @@ func TestGuardian(t *testing.T) {
-			setupHTTPRecordings(t)
+		t.Run("WebAuthn Roaming", func(t *testing.T) {
+			t.Run("Enable", func(t *testing.T) {
+				setupHTTPRecordings(t)
 
-			initialStatus, err := getInitialMFAStatus("webauthn-roaming")
-			assert.NoError(t, err)
+				initialStatus, err := getInitialMFAStatus("webauthn-roaming")
+				assert.NoError(t, err)
 
-			t.Cleanup(func() {
-				err := m.Guardian.MultiFactor.WebAuthnRoaming.Enable(initialStatus)
-				require.NoError(t, err)
+				t.Cleanup(func() {
+					err := m.Guardian.MultiFactor.WebAuthnRoaming.Enable(initialStatus)
+					require.NoError(t, err)
+				})
+
+				err = m.Guardian.MultiFactor.WebAuthnRoaming.Enable(true)
+				assert.NoError(t, err)
+				assertMFAIsEnabled(t, "webauthn-roaming")
 			})
+			t.Run("Settings", func(t *testing.T) {
+				setupHTTPRecordings(t)
 
-			err = m.Guardian.MultiFactor.WebAuthnRoaming.Enable(true)
-			assert.NoError(t, err)
-			assertMFAIsEnabled(t, "webauthn-roaming")
+				initialSettings, err := m.Guardian.MultiFactor.WebAuthnRoaming.Read()
+				assert.NoError(t, err)
+				t.Cleanup(func() {
+					err := m.Guardian.MultiFactor.WebAuthnRoaming.Update(initialSettings)
+					require.NoError(t, err)
+				})
+
+				updatedSettings := &MultiFactorWebAuthnSettings{
+					UserVerification: auth0.String("preferred"),
+				}
+				err = m.Guardian.MultiFactor.WebAuthnRoaming.Update(updatedSettings)
+				assert.NoError(t, err)
+
+				actualSettings, err := m.Guardian.MultiFactor.WebAuthnRoaming.Read()
+				assert.NoError(t, err)
+				assert.Equal(t, "preferred", actualSettings.GetUserVerification())
+			})
 		})
 
-		t.Run("WebAuthn Platform Enable", func(t *testing.T) {
-			setupHTTPRecordings(t)
+		t.Run("WebAuthn Platform", func(t *testing.T) {
+			t.Run("Enable", func(t *testing.T) {
+				setupHTTPRecordings(t)
 
-			initialStatus, err := getInitialMFAStatus("webauthn-platform")
-			assert.NoError(t, err)
+				initialStatus, err := getInitialMFAStatus("webauthn-platform")
+				assert.NoError(t, err)
 
-			t.Cleanup(func() {
-				err := m.Guardian.MultiFactor.WebAuthnPlatform.Enable(initialStatus)
-				require.NoError(t, err)
+				t.Cleanup(func() {
+					err := m.Guardian.MultiFactor.WebAuthnPlatform.Enable(initialStatus)
+					require.NoError(t, err)
+				})
+
+				err = m.Guardian.MultiFactor.WebAuthnPlatform.Enable(true)
+				assert.NoError(t, err)
+				assertMFAIsEnabled(t, "webauthn-platform")
 			})
+			t.Run("Settings", func(t *testing.T) {
+				setupHTTPRecordings(t)
 
-			err = m.Guardian.MultiFactor.WebAuthnPlatform.Enable(true)
-			assert.NoError(t, err)
-			assertMFAIsEnabled(t, "webauthn-platform")
+				initialSettings, err := m.Guardian.MultiFactor.WebAuthnPlatform.Read()
+				assert.NoError(t, err)
+				t.Cleanup(func() {
+					err := m.Guardian.MultiFactor.WebAuthnPlatform.Update(initialSettings)
+					require.NoError(t, err)
+				})
+
+				updatedSettings := &MultiFactorWebAuthnSettings{
+					OverrideRelyingParty: auth0.Bool(false),
```

Comment:
Should we include `RelyingPartyIdentifier` and `OverrideRelyingParty` as well?

Reply:
We could, but I'd avoid doing that because those settings can be modified only if the tenant has a verified custom domain set. So we could run the test with the recorded http interactions but if we run it against a real tenant the test will start to break, unless we add complexity to it to skip sending these fields and checking them if we run against a real tenant. I opted to keep the test simple and avoid doing that because of this. Wdyt?